isis/tests/FunctionalTestsJigsaw.cpp

```diff
@@ -0,0 +1,246 @@
+#include <iostream>
+
+#include <QFile>
+#include <QHash>
+#include <QSet>
+#include <QString>
+#include <QStringList>
+#include <QTextStream>
+
+#include "jigsaw.h"
+#include "Angle.h"
+#include "ControlNet.h"
+#include "ControlPoint.h"
+#include "ControlMeasure.h"
+#include "Distance.h"
+#include "FileName.h"
+#include "Fixtures.h"
+#include "Latitude.h"
+#include "Longitude.h"
+#include "SurfacePoint.h"
+#include "TestUtilities.h"
+#include "gmock/gmock.h"
+#include "UserInterface.h"
+#include "CSVReader.h"
+
+using namespace Isis;
+
+static QString APP_XML = FileName("$ISISROOT/bin/xml/jigsaw.xml").expanded();
+
+// Check to see if a QString contains only numeric values.
+bool isNumeric(QString str){
+  QRegExp re("-*\\d*.*\\d*");  // a digit (\d), zero or more times (*)
+  return re.exactMatch(str);
+}
+
+// Testing help function which will compare csv lines
+void compareCsvLine(CSVReader::CSVAxis csvLine, QString headerStr, int initialIndex=0) { 
+  QStringList compareMe = headerStr.split(",");
+  for (int i=initialIndex; i<compareMe.size(); i++) {
+    if (isNumeric(compareMe[i].trimmed())) {
+      EXPECT_NEAR(csvLine[i].toDouble(), compareMe[i].toDouble(), 0.000001);
+    }
+    else{
+      EXPECT_EQ(QString(csvLine[i]).toStdString(), compareMe[i].toStdString()); 
+    }
+  }
+};
```

Comment:
Probably move this to Utilities.cpp?

Reply:
Can do. Didn't do initially because I wasn't sure how useful they would be elsewhere.